discord/scheduled_event.py

```diff
@@ -47,14 +69,357 @@
     from .user import User
 
     GuildScheduledEventPayload = Union[BaseGuildScheduledEventPayload, GuildScheduledEventWithUserCountPayload]
+    Week = Literal[1, 2, 3, 4, 5]
+    NWeekday = Tuple[Week, ScheduledEventRecurrenceWeekday]
 
 # fmt: off
 __all__ = (
     "ScheduledEvent",
+    "ScheduledEventRecurrenceRule",
 )
 # fmt: on
 
 
+class ScheduledEventRecurrenceRule:
```

Comment:
I feel like the API for this is better off being revised to work similar to `dateutils.rrule` module, or at least similar enough to work like it. Right now this doesn't feel good at a glance, though this is just an initial read on it.

Reply:
Yeah I tried to make it as similar to it as possible, I will rework it soon.